src/draw/lv_image_decoder.c

```diff
@@ -233,6 +258,36 @@ void lv_image_decoder_set_close_cb(lv_image_decoder_t * decoder, lv_image_decode
     decoder->close_cb = close_cb;
 }
 
+void lv_image_decoder_set_cache_free_cb(lv_image_decoder_t * decoder, lv_cache_free_cb_t cache_free_cb)
+{
+    decoder->cache_free_cb = cache_free_cb;
+}
+
+#if LV_CACHE_DEF_SIZE > 0
+lv_cache_entry_t * lv_image_decoder_add_to_cache(lv_image_decoder_t * decoder,
+                                                 lv_image_cache_data_t * search_key,
+                                                 const lv_draw_buf_t * decoded, void * user_data)
+{
+    lv_cache_entry_t * cache_entry = lv_cache_add(img_cache_p, search_key, NULL);
```

Comment:
`search_key` should be deep copied to cache entry because if image path is used as key, it's normally on stack.

Precisely, the issue will happen on lvgl/src/misc/lv_rb.c:88 where the pointer of `key` is stored to node. Which may become wild pointer.

The copy of `key` is also a shallow copy, thus `src` pointer still can become wild. 
https://github.com/lvgl/lvgl/blob/2deef84ff77ffcae89c6c3fb8467e050765ec0b9/src/misc/cache/_lv_cache_lru_rb.c#L92C5-L92C5

Reply:
All of the scenarios where we use the path key now involve making a new copy of the memory, and then freeing the path key's memory at release time, which is how the code in the current project operates. So don't worry about that, where pointers are concerned, everyone makes a good advance copy.

Reply:
So that means user need to make sure the `key` is not on stack and always make a deep copy before adding to cache. Instead of `everyone make an advance copy`, cache manager should do it.

Reply:
It works like [this](https://github.com/lvgl/lvgl/blob/master/src/draw/lv_draw_label.c#L87-L90) in `lv_draw_label_dsc_t` where the text can be local or const too.